odbc-api/src/connection.rs

```diff
@@ -240,6 +245,26 @@ impl<'c> Connection<'c> {
     pub unsafe fn promote_to_send(self) -> force_send_sync::Send<Self> {
         force_send_sync::Send::new(self)
     }
+
+    /// Returns the maximum length of catalog names in bytes.
+    pub fn max_catalog_name_len(&self) -> Result<usize, Error> {
+        self.connection.max_catalog_name_len().map(|v| v as usize)
+    }
+
+    /// Returns the maximum length of schema names in bytes.
+    pub fn max_schema_name_len(&self) -> Result<usize, Error> {
```

Comment:
We use `usize` for other max string lengths (for example, `BufferKind::Text { max_str_len: usize }`) so I thought it might be better to be consistent.

e.g. I plan to use it like this:

```
BufferDescription {
    kind: BufferKind::Text {
        max_str_len: self.max_schema_name_len()?,
    },
    nullable: true,
};
```

so in this case the caller wouldn't need to cast it.

I don't have a preference though -- either seems ok to me.

Reply:
I am somewhat undecided myself, ODBC itself seems to be not very consistent with this either, yet that is mostly due to 
a single signature having to accommodate many different use cases. Personally I would prefer to communicate to the user at this point that he/she has to worry about `u16::max` at the most. Yet I can get behind your reasoning also. I leave the decision up to you. I'll merge this tomorrow morning.

Reply:
Yeah I agree it would be nice to communicate the limits, although I think I'll leave it as `usize`.

Other use cases would probably use this limit for slicing/indexing which need `usize` anyway (e.g. truncating a string at the nearest character boundary with something like `&schema_name[...max_schema_name_len]`), so the slightly improved ergonomics of `usize` are probably worth it in this case.